diff --git a/dlls/box64cpu/cpu.c b/dlls/box64cpu/cpu.c
index 3452b09..8568294 100644
--- a/dlls/box64cpu/cpu.c
+++ b/dlls/box64cpu/cpu.c
@@ -57,28 +57,28 @@ WINE_DECLARE_DEBUG_CHANNEL(box64dump);
 
 NTSTATUS WINAPI Wow64SystemServiceEx( UINT num, UINT *args );
 
-int box64_dynarec_log = 0;
-int box64_dynarec = 1;
-int box64_dynarec_dump = 0;
-int box64_dynarec_forced = 0;
-int box64_dynarec_bigblock = 1;
-int box64_dynarec_forward = 128;
-int box64_dynarec_strongmem = 0;
-int box64_dynarec_x87double = 0;
-int box64_dynarec_fastnan = 1;
-int box64_dynarec_fastround = 1;
-int box64_dynarec_safeflags = 1;
-int box64_dynarec_callret = 0;
-int box64_dynarec_hotpage = 0;
-int box64_dynarec_fastpage = 0;
-int box64_dynarec_bleeding_edge = 1;
-int box64_dynarec_jvm = 1;
-int box64_dynarec_wait = 1;
-int box64_dynarec_test = 0;
-int box64_dynarec_missing = 0;
-int box64_dynarec_trace = 0;
-int box64_dynarec_aligned_atomics = 0;
-int box64_dynarec_div0 = 0;
+int box64_dynarec_log = getenv("BOX64_DYNAREC_LOG") ? atoi(getenv("BOX64_DYNAREC_LOG")) : 0;
+int box64_dynarec = getenv("BOX64_DYNAREC") ? atoi(getenv("BOX64_DYNAREC")) : 1;
+int box64_dynarec_dump = getenv("BOX64_DYNAREC_DUMP") ? atoi(getenv("BOX64_DYNAREC_DUMP")) : 0;
+int box64_dynarec_forced = getenv("BOX64_DYNAREC_FORCED") ? atoi(getenv("BOX64_DYNAREC_FORCED")) : 0;
+int box64_dynarec_bigblock = getenv("BOX64_DYNAREC_BIGBLOCK") ? atoi(getenv("BOX64_DYNAREC_BIGBLOCK")) : 1;
+int box64_dynarec_forward = getenv("BOX64_DYNAREC_FORWARD") ? atoi(getenv("BOX64_DYNAREC_FORWARD")) : 128;
+int box64_dynarec_strongmem = getenv("BOX64_DYNAREC_STRONGMEM") ? atoi(getenv("BOX64_DYNAREC_STRONGMEM")) : 0;
+int box64_dynarec_x87double = getenv("BOX64_DYNAREC_X87DOUBLE") ? atoi(getenv("BOX64_DYNAREC_X87DOUBLE")) : 0;
+int box64_dynarec_fastnan = getenv("BOX64_DYNAREC_FASTNAN") ? atoi(getenv("BOX64_DYNAREC_FASTNAN")) : 1;
+int box64_dynarec_fastround = getenv("BOX64_DYNAREC_FASTROUND") ? atoi(getenv("BOX64_DYNAREC_FASTROUND")) : 1;
+int box64_dynarec_safeflags = getenv("BOX64_DYNAREC_SAFEFLAGS") ? atoi(getenv("BOX64_DYNAREC_SAFEFLAGS")) : 1;
+int box64_dynarec_callret = getenv("BOX64_DYNAREC_CALLRET") ? atoi(getenv("BOX64_DYNAREC_CALLRET")) : 0;
+int box64_dynarec_hotpage = getenv("BOX64_DYNAREC_HOTPAGE") ? atoi(getenv("BOX64_DYNAREC_HOTPAGE")) : 0;
+int box64_dynarec_fastpage = getenv("BOX64_DYNAREC_FASTPAGE") ? atoi(getenv("BOX64_DYNAREC_FASTPAGE")) : 0;
+int box64_dynarec_bleeding_edge = getenv("BOX64_DYNAREC_BLEEDING_EDGE") ? atoi(getenv("BOX64_DYNAREC_BLEEDING_EDGE")) : 1;
+int box64_dynarec_jvm = getenv("BOX64_DYNAREC_JVM") ? atoi(getenv("BOX64_DYNAREC_JVM")) : 1;
+int box64_dynarec_wait = getenv("BOX64_DYNAREC_WAIT") ? atoi(getenv("BOX64_DYNAREC_WAIT")) : 1;
+int box64_dynarec_test = getenv("BOX64_DYNAREC_TEST") ? atoi(getenv("BOX64_DYNAREC_TEST")) : 0;
+int box64_dynarec_missing = getenv("BOX64_DYNAREC_MISSING") ? atoi(getenv("BOX64_DYNAREC_MISSING")) : 0;
+int box64_dynarec_trace = getenv("BOX64_DYNAREC_TRACE") ? atoi(getenv("BOX64_DYNAREC_TRACE")) : 0;
+int box64_dynarec_aligned_atomics = getenv("BOX64_DYNAREC_ALIGNED_ATOMICS") ? atoi(getenv("BOX64_DYNAREC_ALIGNED_ATOMICS")) : 0;
+int box64_dynarec_div0 = getenv("BOX64_DYNAREC_DIV0") ? atoi(getenv("BOX64_DYNAREC_DIV0")) : 0;
 
 int arm64_atomics = 0;
 int arm64_crc32 = 0;
